authorization/role/repository/default_role_mapping.go

```diff
@@ -0,0 +1,202 @@
+	}
+	return exists, nil
+}
+
+// CRUD Functions
+
+// Load returns a single RoleMapping as a Database Model
+func (m *GormDefaultRoleMappingRepository) Load(ctx context.Context, id uuid.UUID) (*DefaultRoleMapping, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "default_role_mapping", "load"}, time.Now())
+	var native DefaultRoleMapping
+	err := m.db.Table(m.TableName()).
+		Preload("ResourceType").
+		Preload("FromRole").
+		Preload("ToRole").
+		Where("default_role_mapping_id = ?", id).Find(&native).Error
+	if err == gorm.ErrRecordNotFound {
+		return nil, errors.NewNotFoundError("default_role_mapping", id.String())
+	}
+	return &native, errs.WithStack(err)
+}
+
+// Create creates a new record.
+func (m *GormDefaultRoleMappingRepository) Create(ctx context.Context, u *DefaultRoleMapping) error {
+	defer goa.MeasureSince([]string{"goa", "db", "default_role_mapping", "create"}, time.Now())
+	if u.DefaultRoleMappingID == uuid.Nil {
+		u.DefaultRoleMappingID = uuid.NewV4()
+	}
+	err := m.db.Create(u).Error
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"default_role_mapping_id": u.DefaultRoleMappingID,
+			"err": err,
+		}, "unable to create the default role mapping")
+		return errs.WithStack(err)
+	}
+	log.Debug(ctx, map[string]interface{}{
+		"default_role_mapping_id": u.DefaultRoleMappingID,
+	}, "Default role mapping created!")
+	return nil
+}
+
+// Save modifies a single record
+func (m *GormDefaultRoleMappingRepository) Save(ctx context.Context, model *DefaultRoleMapping) error {
+	defer goa.MeasureSince([]string{"goa", "db", "default_role_mapping", "save"}, time.Now())
+
+	obj, err := m.Load(ctx, model.DefaultRoleMappingID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"default_role_mapping_id": model.DefaultRoleMappingID,
+			"err": err,
+		}, "unable to update default role mapping")
+		return errs.WithStack(err)
+	}
+	err = m.db.Model(obj).Select("ResourceTypeID", "FromRoleID", "ToRoleID").Updates(model).Error
+	if err != nil {
+		return errs.WithStack(err)
+	}
+
+	log.Debug(ctx, map[string]interface{}{
+		"default_role_mapping_id": model.DefaultRoleMappingID,
+	}, "Default role mapping saved!")
+	return nil
+}
+
+// List returns all default role mappings
+func (m *GormDefaultRoleMappingRepository) List(ctx context.Context) ([]DefaultRoleMapping, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "default_role_mapping", "list"}, time.Now())
+	var rows []DefaultRoleMapping
+
+	err := m.db.Model(&DefaultRoleMapping{}).Find(&rows).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, errs.WithStack(err)
+	}
+	return rows, nil
+}
+
+// Delete removes a single record.
+func (m *GormDefaultRoleMappingRepository) Delete(ctx context.Context, id uuid.UUID) error {
+	defer goa.MeasureSince([]string{"goa", "db", "default_role_mapping", "delete"}, time.Now())
+
+	obj := DefaultRoleMapping{DefaultRoleMappingID: id}
+
+	err := m.db.Delete(&obj).Error
+
+	if err != nil {
```

Comment:
Return NotFoundError if nothing was deleted - https://github.com/fabric8-services/fabric8-auth/blob/master/account/repository/identity.go#L258
And add a test for this.

Reply:
Done